### PR TITLE
Publish to Releases

### LINKED ISSUE #13: Deploy to releases
Provide downloadable binaries for Linux, Windows and MacOS, for those not using cargo.

### PR DESCRIPTION
**Pull request by mlange-42**
* Add workflow to publish binaries to releases
* More flexible tag pattern

Fixes #13

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v0.*.*"
+
+env:
+  CARGO_TERM_COLOR: always
+  REPO: mdbook-toc
```
Comment (francisco-perez-sorrosal) [inline .github/workflows/release.yml:10]: You mean `mdbook-bib`?

Comment (mlange-42) [reply .github/workflows/release.yml:10]: Oops... 😊

Comment (francisco-perez-sorrosal) [reply .github/workflows/release.yml]: :)

**Review by francisco-perez-sorrosal — APPROVED** [review]

**Comment by francisco-perez-sorrosal** [comment]
+1

**Pull request merged**